webapi/src/main/scala/org/knora/webapi/util/standoff/StandoffTagUtilV1.scala

```diff
@@ -0,0 +1,836 @@
+                    }
+
+
+            }.toList
+
+            val attrsGroupedByPropIri: Map[IRI, Seq[StandoffTagAttributeV1]] = attrs.groupBy(attr => attr.standoffPropertyIri)
+
+            // filter all the required props
+            val mustExistOnce: Set[IRI] = classSpecificProps.filter {
+                case (propIri, card) =>
+                    card == Cardinality.MustHaveOne || card == Cardinality.MustHaveSome
+            }.keySet
+
+            // check if all the min cardinalities are respected
+            mustExistOnce.foreach {
+                propIri =>
+                    attrsGroupedByPropIri.get(propIri) match {
+                        case Some(attrs: Seq[StandoffTagAttributeV1]) => ()
+
+                        case None => throw BadRequestException(s"the min cardinalities were not respected for $propIri")
+                    }
+            }
+
+            // filter all the props that have a limited occurrence
+            val mayExistOnce = classSpecificProps.filter {
+                case (propIri, card) =>
+                    card == Cardinality.MustHaveOne || card == Cardinality.MayHaveOne
+            }.keySet
+
+            // check if all the max cardinalities are respected
+            mayExistOnce.foreach {
+                propIri =>
+                    attrsGroupedByPropIri.get(propIri) match {
+                        case Some(attrs: Seq[StandoffTagAttributeV1]) =>
+                            if (attrs.size > 1) {
+                                throw BadRequestException(s"the max cardinalities were not respected for $propIri")
+                            }
+                        case None => ()
+                    }
+            }
+
+            attrs
+
+        } else {
+            // only system props are required
+
+            // check that there no other attributes than datatype attributes and 'class'
+            val unsupportedAttributes: Set[String] = standoffNodeFromXML.attributes.filterNot {
+                attr => (XMLtoStandoffMapping.dataType.nonEmpty && XMLtoStandoffMapping.dataType.get.dataTypeXMLAttribute == attr.key) || attr.key == "class"
+            }.map(attr => attr.xmlNamespace.getOrElse("") + ":" + attr.key + "=" + attr.value)
+
+            if (unsupportedAttributes.nonEmpty) throw BadRequestException(s"Attributes found that are not defined in the mapping: ${unsupportedAttributes.mkString(", ")}")
+
+            Seq.empty[StandoffTagAttributeV1]
+        }
+
+    }
+
+    /**
+      * Represents a text with standoff markup including the mapping.
+      *
+      * @param text          the text as a mere sequence of characters.
+      * @param standoffTagV1 the text's standoff markup.
+      */
+    case class TextWithStandoffTagV1(text: String, standoffTagV1: Seq[StandoffTagV1], mapping: GetMappingResponseV1)
```

Comment:
Please rename this `TextWithStandoffTagsV1`, otherwise it sounds like it contains only one tag.